src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -2826,10 +2846,47 @@ public GlowBlock getDigging() {
         return digging;
     }
 
+    private void broadcastBlockBreakAnimation(GlowBlock block, int destroyStage) {
+        if (block == null) return;
+
+        GlowChunk.Key key = new GlowChunk.Key(block.getChunk().getX(), block.getChunk().getZ());
+        // TODO: should canSeeChunk filtering take place here, in GlowPlayer, or both?
+        block.getWorld().getRawPlayers().stream().filter(player1 -> player1.canSeeChunk(key)).forEach(player2 -> {
+            if (player2 != this) {
+                player2.sendBlockBreakAnimation(player2, block.getLocation(), destroyStage);
+            }
+        });
+    }
+
     public void setDigging(GlowBlock block) {
+        if (block == null) {
+            // remove the animation
+            broadcastBlockBreakAnimation(digging, 10);
+            diggingStarted = -1;
+        } else {
+            // show other clients the block is beginning to crack
+            broadcastBlockBreakAnimation(block, 0);
+            diggingStarted = System.currentTimeMillis();
+        }
+
         digging = block;
     }
 
+    private void pulseDigging() {
+        float hardness = digging.getMaterialValues().getHardness() * 1000;
+
+        // TODO: take into account the tool used to mine (ineffective=5x, effective=1.5x, material multiplier, etc.)
+        // for now, assuming hands are used and the block is not dropped
+        hardness *= 5;
+
+        long duration = System.currentTimeMillis() - diggingStarted;
```

Comment:
Wouldn't it be better to base this system off world thread ticking, instead of system time in milliseconds?